java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/build/preprocessor/AbstractPreprocessor.java

```diff
@@ -1,21 +1,83 @@
  * Some of these may be ignored by different processors
  */
 public abstract class AbstractPreprocessor implements Runnable {
 
     @CommandLine.Parameters(description = "The directory to process")
     protected Path buildRoot;
 
-    @CommandLine.Option(names = { "-dp", "--disable-plugin" }, paramLabel = "<plugin>", description = "The plugin to disable")
+    @CommandLine.Option(names = { "-dp", "--disabled-plugins" }, paramLabel = "<plugin>", description = "The plugin to disable", split=",")
     protected List<String> disabledPlugins;
+
+    @CommandLine.Option(names = "--recipe-image", required = true)
+    String recipeImage;
+
+    @CommandLine.Option(names = "--request-processor-image", required = true)
+    String buildRequestProcessorImage;
+
+    protected enum ToolType {
+        ANT,
+        GRADLE,
+        MAVEN,
+        SBT
+    }
+
+    protected ToolType type;
+
+    @Override
+    public void run() {
+        Path jbsDirectory = Path.of(buildRoot.toString(), ".jbs");
+        //noinspection ResultOfMethodCallIgnored
+        jbsDirectory.toFile().mkdirs();
+
+        String containerFile = """
+            FROM %s
+            USER 0
+            WORKDIR /var/workdir
+            RUN mkdir -p /var/workdir/software/settings /original-content/marker
+            ARG CACHE_URL=""
+            ENV CACHE_URL=$CACHE_URL
+            COPY .jbs/run-build.sh /var/workdir
+            COPY . /var/workdir/workspace/source/
+            RUN /var/workdir/run-build.sh""".formatted(recipeImage);
+
+        // TODO: This is a bit of a hack but as Ant doesn't deploy and the previous implementation relied upon using the
+        //     BuildRequestProcessorImage we need to modify the Containerfile. In future the ant-build.sh should probably
+        //     encapsulate this.
+        if (type == ToolType.ANT) {
+            // Don't think we need to mess with keystore as copy-artifacts is simply calling copy commands.
+            containerFile +=
+                """
+                    FROM %s AS build-request-processor
+                    USER 0
+                    WORKDIR /var/workdir
+                    COPY --from=0 /var/workdir/ /var/workdir/
+                    RUN /opt/jboss/container/java/run/run-java.sh copy-artifacts --source-path=/var/workdir/workspace/source --deploy-path=/var/workdir/workspace/artifacts
+                    FROM scratch
+                    COPY --from=1 /var/workdir/workspace/artifacts /""".formatted(buildRequestProcessorImage);
+        } else {
+            containerFile +=
+                """
+                    FROM scratch
+                    COPY --from=0 /var/workdir/workspace/artifacts /""";
+        }
+        try {
+            Files.writeString(Paths.get(jbsDirectory.toString(), "Containerfile"), containerFile);
+        } catch (IOException e) {
+            Log.errorf("Unable to write Containerfile", e);
+            throw new RuntimeException(e);
+        }
```

Comment:
This inlines the Containerfile generation into the source directory for the next steps (the git archiving and image generation) in the task to use.